### PR TITLE
add contribution guide that includes release process

### LINKED ISSUE #64: Add CONTRIBUTING.md
This repo should contain a document describing how to contribute to `up` and what the current process is for design and implementation of features.

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

Add a contribution guide with release process.

Fixes #64 

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

N/A

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   of all repos to begin "feature freeze" period.
+1. **pin dependencies**: Update the go module on main development branch to
+   depend on stable versions of dependencies if needed.
+1. **branch repo**: Create a new release branch using the GitHub UI for the
+   repo.
+1. **release branch prep**: Make any release-specific updates on the release
+   branch (typically documentation).
+1. **tag release**: Run the `Tag` action on the _release branch_ with the
+   desired version (e.g. `v0.14.0`).
+1. **build/publish**: Run the `CI` and `Configurations` action on the release
```
Comment (hasheddan) [inline CONTRIBUTING.md:31]:
```suggestion
1. **build/publish**: Run the `CI` and action on the release
```

**Review by hasheddan — COMMENTED** [review]
Thanks @muvaf! One comment

File: CONTRIBUTING.md (hunk 1 above)
Comment (muvaf) [reply CONTRIBUTING.md]: Fixed, thanks for catching!

**Review by hasheddan — APPROVED** [review]
Thanks @muvaf!

**Comment by jastang** [comment]
Fixes #64

**Pull request merged**